The fuzz tests were added for the tensorflow::core::function::Runtime class

OleksiiZaderykhinCapgemini opened the pull request:
The fuzz tests were added for the tensorflow::core::function::Runtime class.

OleksiiZaderykhinCapgemini committed:
The fuzz tests were added for the tensorflow::core::function::Runtime class for CreateFunction(const FunctionDef& fdef) and CallFunction(..) methods.

apach301 committed:
Fix crash in HloParser::ParseDenseLiteral when parsing an unexpected token

Add HloParserTest.LiteralDimensionsError test

OleksiiZaderykhinCapgemini committed:
The fuzz tests were added for the tensorflow::core::function::Runtime class.

OleksiiZaderykhinCapgemini committed:
The TransformFunctionFuzzTest::TestPassMock class was added.

Konstantin-Beluchenko commented on tensorflow/security/fuzzing/core/function/runtime_client_fuzz.cc:
```diff
@@ -0,0 +1,242 @@
+
+void CreateFunctionFuzz(FunctionDef def) {
+  auto ctx = LocalEagerCtx();
+  Runtime rt(*ctx);
```
is Runtime creation time-consuming operation?
if so maybe there is sense to move it to the test fixture?

OleksiiZaderykhinCapgemini replied:
The context here holds the current state. That state is denote if the function we create is present in the context or isn't. So the creation of the context is part of the test, and so is the creation of the Runtime.

fcoUnda reviewed (changes requested):
Please use the macro `tf_cc_fuzz_test` found in `tensorflow/security/fuzzing/tf_fuzzing.bzl` instead of tf_cc_shared_test. This macro automatically includes "@com_google_fuzztest//fuzztest" and "@com_google_fuzztest//fuzztest:fuzztest_gtest_main", so you need to remove them from the deps list. You can also remove "@com_google_googletest//:gtest_main", since it is not needed.

OleksiiZaderykhinCapgemini committed:
The `tf_cc_fuzz_test` macro was used.

OleksiiZaderykhinCapgemini commented:
> Please use the macro `tf_cc_fuzz_test` found in `tensorflow/security/fuzzing/tf_fuzzing.bzl` instead of tf_cc_shared_test. This macro automatically includes "@com_google_fuzztest//fuzztest" and "@com_google_fuzztest//fuzztest:fuzztest_gtest_main", so you need to remove them from the deps list. You can also remove "@com_google_googletest//:gtest_main", since it is not needed.

@fcoUnda, Done

nSircombe committed:
[oneDNN][AArch64] Updates oneDNN and Compute Library versions for AArch64 build.

For AArch64 build with oneDNN+ACL (--config=mkl_aarch64 or mkl_aarch64_threadpool)
- Updates oneDNN to v2.7.3 release
- Updates Compute Library (ACL) to v22.11 release
- Removes two patches to ACL which are no longer required
-- tensorflow/third_party/compute_library/acl_depthwise_updateable_weights.patch
-- tensorflow/third_party/compute_library/acl_fixup_SVE_merges.patch
- Adds a new patch to fix minor issue in OpenMP build (addressed in next
  ACL release).
-- tensorflow/third_party/compute_library/acl_openmp_fix.patch

jszaday committed:
Add bounds-checking to `TPUPartitionedInput` and `TPUPartitionedOutput` ops.

PiperOrigin-RevId: 502936861

Commit:
[THLO] Remove unused function `fuseConcatenateOpThroughPointRecursively`

PiperOrigin-RevId: 502944911

chandrasekhard2 committed:
Replace include path for files depending on files in gRPC's `grpcpp/impl/codegen` directory to their corresponding ones in non-codegen directories

PiperOrigin-RevId: 502946732

angerson committed:
Remove "deprecation" language

"Removed" or "redundant" is more accurate.

PiperOrigin-RevId: 502955998

Commit:
[KernelGen] Add experimental JIT-compiled i64-indexed GPU kernels for trigonometric ops

PiperOrigin-RevId: 502959649

tensorflower-gardener committed:
Removes CUDA dependencies from Windows build.

PiperOrigin-RevId: 502960161

Commit:
[THLO] Generate tiled implementation for thlo.concat based on a single operand if possible

If the requested tile is of size 1 in the concatenation dimensions, we can extract the tile from one of the operands.
In that case, there is no need to generate a tiled thlo.concat operation.
Rather, we can extract the requested tile from the unique operand directly.
This special form of tiling applied, in particular, to unit tiles, i.e. scalars.

PiperOrigin-RevId: 502960672

tensorflower-gardener committed:
Updates Bazel version of TensorFlow build scripts.

PiperOrigin-RevId: 502963284

tensorflower-gardener committed:
Merge pull request #59253 from cfRod:onednn_acl_threadpool_fix

PiperOrigin-RevId: 502960791

tensorflower-gardener committed:
Merge pull request #58862 from Intel-tensorflow:sriniva2/pooling_v1_fix

PiperOrigin-RevId: 502964144

zhangqiaorjc committed:
Store sorted flattened dict keys in PyTree as a c++ vector instead of py::list to avoid creating new python object on every single dict flatten. For deeply nested dict, this avoids excessive gc pressure and avoids the slowdown whenever gc needs to sweep too many live python objects.

PiperOrigin-RevId: 502967020

fergushenderson committed:
Use cc_library_with_tflite for sample_stable_delegate_external.
Also, fix a type error where it was using TfLiteDelegate instead of TfLiteOpaqueDelegate.

PiperOrigin-RevId: 502970231

tensorflower-gardener committed:
Merge pull request #59233 from yongtang:59160-stateless_random_binomial-crash

PiperOrigin-RevId: 502968995

tensorflower-gardener committed:
Merge pull request #59292 from Intel-tensorflow:collective_test_fix

PiperOrigin-RevId: 502976977

Commit:
[SE] Replace SE status macros/functions with TF macros/functions.

These are the same.  Just renaming.

PiperOrigin-RevId: 502978759

yangustc07 committed:
#tf-data-service Distributed snapshot task def supports compression.

PiperOrigin-RevId: 502978855

tensorflower-gardener committed:
Rollback of PR #59233

PiperOrigin-RevId: 502989604

yangustc07 committed:
#tf-data-service Do not create an interator if a writer already exists.

PiperOrigin-RevId: 502992305

sagunb committed:
Fix an issue with the possible re-use of the function_def pointer in the calling context.

PiperOrigin-RevId: 502992585

Commit:
[SE] Replace SE status macros/functions with TF macros/functions.

These are the same.  Just renaming.

PiperOrigin-RevId: 502997055

tensorflower-gardener committed:
Merge pull request #59148 from nluehr:gpu-host-mem-limit

PiperOrigin-RevId: 502996189

LukeBoyer committed:
Label `tfl.quantize` and `tfl.dequantize` as having `NoMemoryEffect` in their dialect definitions. This permits them to be removed by `applyPatternsAndFoldGreedily` when they are unused. This solves the issue of hanging dequant nodes.

PiperOrigin-RevId: 503002681

philipphack committed:
Support for batched FP8 GEMMs in XLA.

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@198d1d997697

Updates LLVM usage to match
[198d1d997697](https://github.com/llvm/llvm-project/commit/198d1d997697)

PiperOrigin-RevId: 503012077

changm committed:
Move up StableHLO pass to before the first legalization. Needed for b/188389290. Some tests were failing since legalization for StableHLO ops. Moving to the beginning since it seems to not matter where we lower StableHLO.

Also seems to make sense to make it first so that we immediately always have MHLO.

PiperOrigin-RevId: 503012601

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/e773a2f0357ced66a873b127a6e6dd70d28fd03d.

PiperOrigin-RevId: 503014899

BrianWieder committed:
Remove autograph/util's unneeded dependency on tf.data

PiperOrigin-RevId: 503017803

tensorflower-gardener committed:
Copy quantization_config for tpu_embedding_layer

PiperOrigin-RevId: 503056874

tensorflower-gardener committed:
Merge pull request #59246 from nluehr:cuda-set-device-fix

PiperOrigin-RevId: 503069642

tensorflower-gardener committed:
Merge pull request #58284 from jglaser:fix_rocblas_atomics

PiperOrigin-RevId: 503085218

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-01-19

PiperOrigin-RevId: 503090518

tensorflower-gardener committed:
Update GraphDef version to 1381.

PiperOrigin-RevId: 503090519

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503093517

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@38a2f089b481

Updates LLVM usage to match
[38a2f089b481](https://github.com/llvm/llvm-project/commit/38a2f089b481)

PiperOrigin-RevId: 503107364

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/0640150b6b163ca47e8dce8e67d012df0af55365.

PiperOrigin-RevId: 503109609

tensorflower-gardener committed:
Use new fuzz macro for cc tests.

PiperOrigin-RevId: 503145510

chr1sj0nes committed:
[STREAM_EXEC] Change `ThenDoHostCallback[WithStatus]` to take an `absl::AnyInvocable` callback.

This removes the need for the callback to copyable.

PiperOrigin-RevId: 503161544

tensorflower-gardener committed:
[XLA:GPU] Use a global thread pool in ShapeUtil::ForEachIndexInternalParallel

We got reports that always recreating the threads were slow.

We used the same global thread pool pattern as cuda_driver.cc/GetDriverExecutor.
https://github.com/tensorflow/tensorflow/blob/7ab78a5d7721ec10f3a742bb8f997ee34c2afe15/tensorflow/compiler/xla/stream_executor/cuda/cuda_driver.cc#L102

PiperOrigin-RevId: 503169365

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@94750af83640

Updates LLVM usage to match
[94750af83640](https://github.com/llvm/llvm-project/commit/94750af83640)

PiperOrigin-RevId: 503171684

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/3d1aebfcffa90eeec7f678349ab9cadf47a180bc.

PiperOrigin-RevId: 503174500

tensorflower-gardener committed:
Merge pull request #58795 from trevor-m:tmorris-bf16-scan

PiperOrigin-RevId: 503174586

tensorflower-gardener committed:
Propagate annotation `_parallel_group_ids` in `LaunchToDeviceAttributePass`.

This annotation will be used later in `UpdateControlDependenciesPass`

PiperOrigin-RevId: 503180681

Commit:
Support module-side stable delegate registration and retrieval.

PiperOrigin-RevId: 503182390

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 503198842

philipphack committed:
Support for batched FP8 GEMMs in XLA.

ezhulenev committed:
[tensorflow] Fix OSS stablehlo + Tensorflow build

PiperOrigin-RevId: 503199131

changm committed:
Remove external function for CreateConvertMlirToXlaHloPipeline and move the
function into the anon namespace to prevent leakage.

PiperOrigin-RevId: 503203877

tensorflower-gardener committed:
Internal unit tests only.

PiperOrigin-RevId: 503209722

ezhulenev committed:
[xla:gpu] Wait for the completion of all memcpy operation in ResolveConstantGlobals

Context synchronization is a too big hammer for the problem at hand, and leads to crashes if some other binary happened to be in CUDA graph capture mode. Explicitly wait for the completion of all memory transfers in global constant initializer.

PiperOrigin-RevId: 503210736

tensorflower-gardener committed:
Merge pull request #59142 from trevor-m:tmorris-bf16-segment-reduction

PiperOrigin-RevId: 503211129

tensorflower-gardener committed:
Propagate `tensorflow.SessionMetadata` in `EagerContext`

PiperOrigin-RevId: 503214970

tensorflower-gardener committed:
Merge pull request #59304 from Jerry-Ge:master

PiperOrigin-RevId: 503211719

tensorflower-gardener committed:
Merge pull request #59226 from pollfly:patch-1

PiperOrigin-RevId: 503212073

BrianWieder committed:
Add DOCKER_HOST_CACHEBUSTER so that host and exec configurations bust cache on a new docker image build.

This should fix the clang builds failing when using cache.

PiperOrigin-RevId: 503218397

hehhjiang committed:
Implement a GPU plugin for minibenchmark that can load gpu plugin from TFLite-in-PlayServices GPU Module.

PiperOrigin-RevId: 503218622

alankelly committed:
Free unused temporary tensors for delegated nodes

PiperOrigin-RevId: 503219385

tensorflower-gardener committed:
Update `ParallelExecuteToIslands` pass

The new GraphExport no longer requires manually adding control deps to the Graph Fetch, since the fetches will be computed using side-effect analysis at the very end of GraphExport. Also, we introduce annotation `_parallel_execution_ids` into the pass.

This will be used in `UpdateControlDependencies` pass in new graph_export_pipeline.

PiperOrigin-RevId: 503219450

d0k committed:
[XLA:CPU:Runtime] Wire up --xla_cpu_enable_mlir_tiling_and_fusion when running with XLA runtime

PiperOrigin-RevId: 503222110

tensorflower-gardener committed:
Refactor DTensorXlaSpmdIntegration to separated passes

PiperOrigin-RevId: 503225920

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503227822

mikedelorimier committed:
Add logging for graph-optimization-like rewrites to be similar to optimization_registry's logging. Also log when the compilation part of InstantiateMultiDevice is finished.

To enable logging for passes under InstantiateMultiDevice use: --vmodule=process_function_library_runtime=2,optimize_function_graph_utils=2,optimization_registry=1,mlir_graph_optimization_pass=2,replicate_per_replica_nodes=1,function_utils=2

PiperOrigin-RevId: 503229379

tensorflower-gardener committed:
Add more debug vlogs.

PiperOrigin-RevId: 503234436

faizan-m committed:
Add experimental_attributes to tf.function

PiperOrigin-RevId: 503234529

smit-hinsu committed:
Reject bounded dynamic inputs for Split and SplitV lowering

If the Split op axis dimension size is odd, then it should return an error but we can't know that at the compile time if the input is bounded.

SplitV op takes split sizes so we could support that assuming that the input size matches the requested sizes. However, if one of the split size is `-1`, then we need to set dynamic dimension appropriate that we currently don't do.

PiperOrigin-RevId: 503234610

smit-hinsu committed:
Move legalize tf control flow patterns to the legalize tf pass

Remove all the usage of the legalize tf control flow pass. All the users also invoke the legalize tf pass before or after the control flow legalization pass.

Patterns and tests are moved without any change except a change in one of the tests. It is now using the tf dialect ops to verify that legalization of ops with regions correctly inter operates with the other patterns.

Legalizing ops with regions with the other ops will allow type conversion during the legalization. That is required for supporting quantization types in control flow ops because the TensorFlow uses qint types and MHLO uses quantization types and neither supports the other one. (Alternative could have been to introduce casts but this is much simpler and all legalization is in a function pass.)

PiperOrigin-RevId: 503236736

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503237336

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503238366

nitins17 committed:
Increase the size limit for Linux CPU wheels

PiperOrigin-RevId: 503240788

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503241511

faizan-m committed:
Instantiate FunctionSpecs from FunctionType

PiperOrigin-RevId: 503242564

Commit:
[TF:PLUGIN] Load pluggable device library when environment variable TF_PLUGGABLE_DEVICE_LIBRARY_PATH is set.

PiperOrigin-RevId: 503249451

tensorflower-gardener committed:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503256489

broken committed:
Add missing types (uint32_t & uint16_t) to available types in TFLite shim's TensorView.

PiperOrigin-RevId: 503258406

tensorflower-gardener committed:
Merge pull request #59278 from nSircombe:feature/github/oneDNN273_ACL2211_update

PiperOrigin-RevId: 503257419

broken committed:
Improve on the Templated TF Lite shim example. Previously, users would need to create their own wrapper class to delegate to the correctly typed op object. With this change, the OpWrapper is created for the user, and they only need to pass in the correct attribute types.

PiperOrigin-RevId: 503260524

cezheng committed:
[PJRT] Add a method to PjRtClient to create an error buffer without allocating memory.

PiperOrigin-RevId: 503264732

haozha111 committed:
Check filter_input_channel > 0 in conv kernel.

PiperOrigin-RevId: 503266928

rohitju committed:
Disable test for sanitizers.

PiperOrigin-RevId: 503271373

tensorflower-gardener committed:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503271574

faizan-m committed:
Make binding failure error message more helpful

PiperOrigin-RevId: 503271919

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503273417

pineapplejuice233 committed:
Ignore the max_seq_length setting when the rank of the input tensor is above 2 for tpu embedding mid level API. Update the serving API logic align with the TPU one.

PiperOrigin-RevId: 503273477

tensorflower-gardener committed:
Make the reference integer fully connected kernel to be generic

PiperOrigin-RevId: 503273939

trevor-m committed:
Add bfloat16 to TF_CALL_GPU_NUMBER_TYPES

qlzh727 committed:
Add a util method to check if a tensor is a DTensor instance.

PiperOrigin-RevId: 503276072

tensorflower-gardener committed:
Merge pull request #55383 from nex-team-inc:personal/tony/better-load-ext-delegate-error

PiperOrigin-RevId: 503278524

JXRiver committed:
[Security] Update int overflow fix in ragged range.

So ASAN won't fail.

PiperOrigin-RevId: 503281076

sirakiin committed:
Instrument hooks for reporting results of ModifyGraphWithDelegate and Invoke.

PiperOrigin-RevId: 503281679

LukeBoyer committed:
Update `TfLiteTensorRealloc` to return a `TfLiteStatus` and error when allocation fails or tensors with inappropriate allocation semantics are passed in.

PiperOrigin-RevId: 503284353

tensorflower-gardener committed:
Reject bounded dynamic inputs for Split and SplitV lowering

If the Split op axis dimension size is odd, then it should return an error but we can't know that at the compile time if the input is bounded.

SplitV op takes split sizes so we could support that assuming that the input size matches the requested sizes. However, if one of the split size is `-1`, then we need to set dynamic dimension appropriate that we currently don't do.

PiperOrigin-RevId: 503284664

yishuangP committed:
Updated Documentation on CoreML delegate. Add new fields in TfLiteCoreMlDelegateOptions for debugging purpose.

PiperOrigin-RevId: 503285467

Commit:
[SE] Remove references to stream_executor's status{,or}.h.

We've removed uses of the types and functions declared in these headers.

PiperOrigin-RevId: 503285602

BrianWieder committed:
Remove `tensorflow/python/autograph/operators`'s dependency on tf.data. This helps to remove incorrect usages of LazyLoader that were used to avoid a cyclic import.

PiperOrigin-RevId: 503285610

hawkinsp committed:
Fix test failures under NumPy 1.24.

NumPy 1.24 release notes: https://numpy.org/devdocs/release/1.24.0-notes.html

The fixes vary, but there are three particularly common changes:
* NumPy 1.24 removes a number of deprecated NumPy type aliases references (np.bool, np.int, np.float, np.complex, np.object, np.str, np.unicode, np.long). This change replaces them with their recommended replacements (bool, int, float, complex, object, str, str, int).
* Under NumPy 1.24 no longer automatically infers dtype=object when ragged sequences are passed to np.array(). See https://numpy.org/neps/nep-0034-infer-dtype-is-object.html . In most cases the fix is to pass dtype=object explicitly, but in some cases where the raggedness seems accidental other fixes were used.
* NumPy 1.24 is pickier about the dtype= option passed to comparison ufuncs.

PiperOrigin-RevId: 503288573

tensorflower-gardener committed:
[xla:runtime] Add custom call test in executable_test

PiperOrigin-RevId: 503288731

tensorflower-gardener committed:
Make the reference integer element wise multiply function generic

PiperOrigin-RevId: 503289550

mpcallanan committed:
Move distributed snapshot fault tolerance tests to their own file.

Also remove the sanitizer blocks from the old test file.

PiperOrigin-RevId: 503294107

ddunl committed:
Add compiler/xla/stream_executor:cuda_platform to deps of run_hlo_module, replay_computation_gpu, and interactive_graphviz.
This fixes errors when run under --config=cuda in OSS

PiperOrigin-RevId: 503296736

ddunl committed:
Remove compiler/xla/stream_executor/lib/demangle.{cc,h} and update user to TSL equivalent

PiperOrigin-RevId: 503298223

tensorflower-gardener committed:
Add unsigned integer types to MatMul and BatchMatMulV2 as supported types.

PiperOrigin-RevId: 503302020

tensorflower-gardener committed:
Merge pull request #59318 from trevor-m:tmorris-bf16-macro

PiperOrigin-RevId: 503304085

tensorflower-gardener committed:
This CL adds a new RewritePattern to check if a BatchMatMulOp, with no broadcasting necessary, can be lowered to FullyConnectedOp. If possible, lowers a TF::BatchMatMulOp to TFL::FullyConnectedOp.

PiperOrigin-RevId: 503305389

mihaimaruseac committed:
Quick fix for a vuln in printing empty tensors.

PiperOrigin-RevId: 503308327

tensorflower-gardener committed:
Ignore the max_seq_length setting when the rank of the input tensor is above 2 for tpu embedding mid level API. Update the serving API logic align with the TPU one.

PiperOrigin-RevId: 503309949

Commit:
[XLA] Do not perform copy optimization of replacing it with an operand when the copy has a sharding applied to it and is the module root.

Input/output sharding are interfaces with the runtime systems. Replacing root instructions without preserving the sharding
is a problem. Avoid doing that for copies here.

PiperOrigin-RevId: 503310848

dansuh17 committed:
Apply the new formatter (pyink) changes to TF Quantization's python codebase.

PiperOrigin-RevId: 503313089

srujun committed:
Add DTensor SPMD expander for IteratorGetNextOp.

This change adds an SPMD expander implementation for IteratorGetNextOp which will enable dataset iterators to be used inside tf.function. The element layouts are attached to the iterator resource DTensor as an attribute, and then used inside the SPMD Expansion algorithm. Iterator resources are created for each host mesh device and packed into a DTensor.

PiperOrigin-RevId: 503313097

yangustc07 committed:
#tf-data-service Disable a flaky test.

PiperOrigin-RevId: 503313415

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 503316512

pschuh committed:
Move xplane_to_trace_events and trace_events_to_json to tsl.

PiperOrigin-RevId: 503320621

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@afca08a567e3

Updates LLVM usage to match
[afca08a567e3](https://github.com/llvm/llvm-project/commit/afca08a567e3)

PiperOrigin-RevId: 503320763

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/00477f56b1b952904e82c8240724676d529b3c31.

PiperOrigin-RevId: 503322682

ezhulenev committed:
[xla:gpu] Use stream executor events to signal Send/Recv operations completion

It's one step in the direction of a fully async Send/Recv operations in XLA:GPU. Currently we are still forced to block the caller thread when doing Send, because SendCallback doesn't allow us to pass not-yet-available chunk.

PiperOrigin-RevId: 503347033

dansuh17 committed:
Implement `type_caster` for `QuantizationOptions` in pybind11.

Instead of manually deserializing QuantizationOptions from the cpp side, uses a type_caster in pybind11 for
converting serialized quantization options into `QuantizationOptions`.

PiperOrigin-RevId: 503347249

ezhulenev committed:
[xla:gpu] Schedule send operations into the PjRtClient thread pool

Remove `BlockHostUntilDone` from the synchronous execution path to enable truly async send/recv operations.

PiperOrigin-RevId: 503354840

akuegel committed:
Move ROCM build to 5.3

Also add python 3.10 and python 3.11

PiperOrigin-RevId: 503358144

tensorflower-gardener committed:
Add unit-test for non-frozen variable quantization

PiperOrigin-RevId: 503371092

tensorflower-gardener committed:
Merge pull request #59296 from Intel-tensorflow:gaurides/onednn2.7.3

PiperOrigin-RevId: 503374863

tensorflower-gardener committed:
Update GraphDef version to 1382.

PiperOrigin-RevId: 503377352

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-01-20

PiperOrigin-RevId: 503377382

jreiffers committed:
MLIR debugger framework and core dialects.

This includes the minimum needed to run some trivial IR (mostly func.call, arith and tensor).

PiperOrigin-RevId: 503398807

jreiffers committed:
Implement the most common memref ops.

PiperOrigin-RevId: 503405877

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@3ac4299d3798

Updates LLVM usage to match
[3ac4299d3798](https://github.com/llvm/llvm-project/commit/3ac4299d3798)

PiperOrigin-RevId: 503406608

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/1ca7d6b5d0aaeff8a42b71ea02467f809abec2ec.

PiperOrigin-RevId: 503408625

jreiffers committed:
Implement affine.{min,max,apply}.

Add some more text so the check sees that it's actually more than three words.

PiperOrigin-RevId: 503409784

MatthiasKreileder committed:
Provide a set of TF_LITE_ENSURE macros that work with the opaque context.  This allows clients that use the stable delegate API to check conditions within their code and report errors if the conditions don't hold.

PiperOrigin-RevId: 503411126

jreiffers committed:
Implement most of the vector dialect.

PiperOrigin-RevId: 503417046

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@02fb5aae1109

Updates LLVM usage to match
[02fb5aae1109](https://github.com/llvm/llvm-project/commit/02fb5aae1109)

PiperOrigin-RevId: 503420578

tensorflower-gardener committed:
Provide a set of TF_LITE_ENSURE macros that work with the opaque context.  This allows clients that use the stable delegate API to check conditions within their code and report errors if the conditions don't hold.

PiperOrigin-RevId: 503421487

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/8f0f0696cb3736d1e62f5aabea88b44ef9f4013a.

PiperOrigin-RevId: 503422707

akuegel committed:
Revert Move ROCM build to 5.3

Also add python 3.10 and python 3.11

PiperOrigin-RevId: 503430840

smit-hinsu committed:
Relaxed the condition to allow dynamic dimensions in non split dimensions

PiperOrigin-RevId: 503435068

jreiffers committed:
Implement some common linalg operations.

broadcast, fill, generic, map, matmul, reduce.

PiperOrigin-RevId: 503439362

fcoUnda committed:
Use new macro

MatthiasKreileder committed:
Provide a set of TF_LITE_ENSURE macros that work with the opaque context.  This allows clients that use the stable delegate API to check conditions within their code and report errors if the conditions don't hold.

PiperOrigin-RevId: 503441429

ashishenoyp committed:
Move optional_ops out of tensorflow/core/ops/dataset_ops.cc into optional_ops.cc

PiperOrigin-RevId: 503446288

tensorflower-gardener committed:
Merge pull request #59202 from yongtang:58175-BlockLSTM-crash

PiperOrigin-RevId: 503451313

pineapplejuice233 committed:
If max_seq_length is set and rank is unknown, assume the input tensor is rank 2.

PiperOrigin-RevId: 503453779

tensorflower-gardener committed:
Internal compatible_with change.

PiperOrigin-RevId: 503457385

jreiffers committed:
Add SCF for, parallel, if, while.

PiperOrigin-RevId: 503458386

jszaday committed:
Enable TPUPartitionedInput to TPUPartitionedInputV2 Conversion Pass.

PiperOrigin-RevId: 503462579

yangustc07 committed:
#tf-data-service Add a class to read distributed snapshots.

This is useful for unit tests for now. Later it can be extended
to implement the distributed `load` op.

PiperOrigin-RevId: 503467987

datenglin committed:
Enhanced the future test by checking error strings.

PiperOrigin-RevId: 503470419

tensorflower-gardener committed:
Use new macro in op fuzzers.

PiperOrigin-RevId: 503472369

yangustc07 committed:
#tf-data-service Move distributed_snapshot_test to snapshot/ directory.

PiperOrigin-RevId: 503473028

fionalang committed:
Create a TypeSpec interface for instance checks as a first step in removing the dependency cycle between `saved_model/nested_structure_coder.py` and `framework/type_spec.py`.

PiperOrigin-RevId: 503473950

tensorflower-gardener committed:
Adds BACKEND_PASSES stage to the `CompilationLogEntry` proto.

PiperOrigin-RevId: 503476779

yangustc07 committed:
#tf-data-service Clean up a forward compatibility check.

`register_dataset` always returns a string now:
https://github.com/tensorflow/tensorflow/blob/42370fcd886be44b105d7e8547a8013c244ee046/tensorflow/python/data/experimental/ops/data_service_ops.py#L836
https://github.com/tensorflow/tensorflow/blob/42370fcd886be44b105d7e8547a8013c244ee046/tensorflow/core/kernels/data/experimental/data_service_ops.cc#L152

PiperOrigin-RevId: 503478016

cantonios committed:
Fix audio spectrogram FPE.

Do input validation in shape function.

PiperOrigin-RevId: 503481241

jreiffers committed:
Minimal bufferization dialect: alloc_tensor, to_tensor, to_memref.

PiperOrigin-RevId: 503482243

mpcallanan committed:
Move distributed snapshot task protos from dispatcher.proto to common.proto.

These make more sense to be accessed from common.proto in a forthcoming CL testing worker<->dispatcher interactions.  Also `TaskDef` is already in common.proto so maybe `SnapshotTaskDef` should be there, too.

PiperOrigin-RevId: 503484754

ezhulenev committed:
[xla:gpu] Add async error handler to Send/Recv functions

PiperOrigin-RevId: 503484942

aaudiber committed:
#tf-data Improve test naming for auto shard test.

PiperOrigin-RevId: 503486810

atondwal committed:
Hotfix for stablehlo CMake

Patches in openxla/stablehlo#949 to hotfix the CMake build.
CMake version of tensorflow/tensorflow@225542e

PiperOrigin-RevId: 503487880

grantjensen committed:
Was passing a bad shape into tf for a constant perm, so these tests weren't getting compared to TFL.

PiperOrigin-RevId: 503490746

tensorflower-gardener committed:
Internal compatible_with change.

PiperOrigin-RevId: 503492937

olegshyshkov committed:
[GmlSt] Use short print form for linalg primitive ops in gml_st tests.

PiperOrigin-RevId: 503495148

tensorflower-gardener committed:
Add an HLO for Tan(x).

PiperOrigin-RevId: 503500163

cantonios committed:
Add validation checks in sparse binary ops.

Fixes a nullptr security vulnerability.

PiperOrigin-RevId: 503501621

anlunx committed:
[XLA:GPU] Implement GEMM runtime autotuning

PiperOrigin-RevId: 503510965

tensorflower-gardener committed:
Fix typos in comments describing FP8 dtypes.

Number of mantissa bits was incorrect.

PiperOrigin-RevId: 503512825

Commit:
[KernelGen] Group rounding kernels in BUILD file

PiperOrigin-RevId: 503512992

tensorflower-gardener committed:
Remove the `estimator_export` decorator.

All uses of `estimator_export` in `tensorflow_estimator` have
been updated to use an alternative decorator defined inside the
estimator package.

As the estimator is a deprecated API, no new endpoints should
be registered using the `estimator_export` decorator. This change
removes the decorator from `tf_export`.

`estimator_export` was also the last remaining usage of the `deprecated_inst`
argument in the overall `api_export` decorator. This optional argument
created a circular dependency between the `deprecation` module and
`tf_export`. This argument and its functionality have also been removed.

PiperOrigin-RevId: 503514596

changm committed:
Boilerplate code to create a verification pass that all TF/XLA ops have been
lowered to MHLO.

Will add actual logic and begin extraction of TF/XLA full legalization checks here in a follow up change.

PiperOrigin-RevId: 503522945

Commit:
[KernelGen] Group predicate kernels in build file

PiperOrigin-RevId: 503523136

Commit:
Clean up references to the protobuf compat library

PiperOrigin-RevId: 503525777

Tongfei-Guo committed:
[XLA:SPMD] Allows propagating Manual sharding in many cases.

PiperOrigin-RevId: 503527130

BrianWieder committed:
Remove usage of LazyLoader to avoid cyclic import when importing `autograph/impl/api.py` in `ops/script_ops.py`.

PiperOrigin-RevId: 503530037

atondwal committed:
Integrate StableHLO at openxla/stablehlo@11bb08a8babb

* Update BatchNorm* op tests for looser verifiers
* Update error message on rng_bit_generator test

PiperOrigin-RevId: 503532701

BrianWieder committed:
Remove usage of LazyLoader to avoid cyclic import when importing `parallel_for/control_flow_ops.py` in `py_builtins.py`.

PiperOrigin-RevId: 503533219

tensorflower-gardener committed:
[tf.data] Add member variables and methods for the buffered bytes and peak buffered bytes in a node.

PiperOrigin-RevId: 503535043

BrianWieder committed:
Remove usage of LazyLoader to avoid cyclic import when importing pfor_ops

PiperOrigin-RevId: 503548777

pschuh committed:
Fix logic for finding first non-zero and slicing information for
https://github.com/google/jax/issues/13824.

The problem is that the code to scan the literal for non-zeros was completely broken. This made it find negative bounds when scanning (because it never would check the final dimension for -1's and simply scan memory arbitrarily until it found a non-zero).
I also fixed how the non-zero checker works for an entirely zero weight matrix. In this case, the bounds would all loop (which is wrong).
Also, when checking the slicing to see if only the feature dim was sliced, the output shape was used instead of the input shape making the check trivially true (slice-dims always equal the input-dims).

PiperOrigin-RevId: 503555714

ghpvnist committed:
Hotfix for breakage

PiperOrigin-RevId: 503558304

fchollet committed:
Create public model.export() API.

PiperOrigin-RevId: 503561685

terryheo committed:
lite: Sync pip package build scripts

Sync tools/pip_package/build_pip_package_with_bazel.sh with tools/pip_package/build_pip_package_with_cmake.sh

PiperOrigin-RevId: 503564793

tensorflower-gardener committed:
Merge pull request #57462 from prince-chrismc:patch-9

PiperOrigin-RevId: 503565747

tensorflower-gardener committed:
Avoid copies for in-place CollectivePermute/CollectivePermuteStart.

PiperOrigin-RevId: 503568764

anlunx committed:
[XLA:GPU] GPU genrule should be able to run without autotune results

This CL also fixes a bug in xla_compile_main.cc

PiperOrigin-RevId: 503573087

Commit:
[XLA] Reorganize HloSharding to save some space.

Alignment of boolean values waste space if in the middle.

PiperOrigin-RevId: 503575290

tensorflower-gardener committed:
Update GraphDef version to 1383.

PiperOrigin-RevId: 503609832

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-01-21

PiperOrigin-RevId: 503609838

tensorflower-gardener committed:
[tf.data] Add a global tfdataz metric registry to record the iterator specific metrics.

PiperOrigin-RevId: 503611385

jreiffers committed:
Implement common mhlo ops.

Significant omissions right now are custom_call (I need to figure out how to
handle dependencies) and concatenate.

PiperOrigin-RevId: 503630046

sirakiin committed:
Report GPU delegate supported node count and which backend is used in GPU delegate kernel.

PiperOrigin-RevId: 503635051

pifon2a committed:
Use tensor.extract_slice instead of gml_st.materialize.

PiperOrigin-RevId: 503656414

ghpvnist committed:
[StableHLO to MHLO] Fix typos in infer_stablehlo.mlir tests

Based on:
https://github.com/openxla/stablehlo/pull/881

PiperOrigin-RevId: 503671076

ghpvnist committed:
[StableHLO to MHLO] Handle bounds in the SelectAndScatterOp shape function

Based on:
https://github.com/openxla/stablehlo/pull/917

PiperOrigin-RevId: 503677683

ghpvnist committed:
[StableHLO to MHLO] Verify exponent and mantissa in ReducePrecisionOp

Based on:
https://github.com/openxla/stablehlo/pull/919

PiperOrigin-RevId: 503677765

jreiffers committed:
Replace 0l with int64_t{0}.

The Mac build doesn't like the former.

PiperOrigin-RevId: 503680843

jreiffers committed:
Hotfix for breakage

PiperOrigin-RevId: 503683038

ghpvnist committed:
[StableHLO to MHLO] Handle bounds in the ReduceWindowOp shape function

Based on:
https://github.com/openxla/stablehlo/pull/916

PiperOrigin-RevId: 503683991

ghpvnist committed:
[StableHLO to MHLO] Handle bounds in the CholeskyOp shape function

Based on:
https://github.com/openxla/stablehlo/pull/887

PiperOrigin-RevId: 503687049

pifon2a committed:
Use tensor.extract_slice instead of gml_st.materialize.

PiperOrigin-RevId: 503689820

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-01-22

PiperOrigin-RevId: 503753717

tensorflower-gardener committed:
Update GraphDef version to 1384.

PiperOrigin-RevId: 503753715

dansuh17 committed:
Add no_pip tag for `pywrap_quantize_model_test`.

PiperOrigin-RevId: 503756917

tensorflower-gardener committed:
Add a new Pass to replicate TensorList init ops.

By replicating the TensorList init ops (undoing some CSE), the shape inference of the assignment through TensorListSetItem can be done without any conflict and then shape_inference pass will be able to resolve it.

PiperOrigin-RevId: 503806125

snadampal committed:
[aarch64] enable acl linking for cpu xla for mkl_aarch64_threadpool configuration

pifon2a committed:
Use tensor.extract_slice instead of gml_st.materialize.

PiperOrigin-RevId: 503820048

tensorflower-gardener committed:
Op documentation update.
	update of g3doc/_includes/tf_passes.md

PiperOrigin-RevId: 503823711

tensorflower-gardener committed:
Add SegmentProdV2op with num_segments as additional input.

The only difference with SegmentProd is the additional input  `num_segment`.
This helps in evaluating the output shape in compile time.

PiperOrigin-RevId: 503827333

tensorflower-gardener committed:
Use tensor.extract_slice instead of gml_st.materialize.

PiperOrigin-RevId: 503828263

MatthiasKreileder committed:
Initialize the telemetry_settings_ member variable.

PiperOrigin-RevId: 503828382

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 503834792

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 503837737

tensorflower-gardener committed:
Add SegmentProdV2op with num_segments as additional input.

The only difference with SegmentProd is the additional input  `num_segment`.
This helps in evaluating the output shape in compile time.

PiperOrigin-RevId: 503846096

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 503849326

tensorflower-gardener committed:
Internal change

PiperOrigin-RevId: 503854263

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 503860818

tensorflower-gardener committed:
Add UnpackQuantMHLO pass that unpacks MHLO quantized ops to MHLO primitive ops.

PiperOrigin-RevId: 503861621

jreiffers committed:
gml_st.for, parallel, materialize, tile.

PiperOrigin-RevId: 503907978

tensorflower-gardener committed:
Merge pull request #59303 from philipphack:u_fp8_batched_gemm_xla

PiperOrigin-RevId: 503910853

tensorflower-gardener committed:
Update GraphDef version to 1385.

PiperOrigin-RevId: 503913277

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-01-23

PiperOrigin-RevId: 503913273

ingomueller-net committed:
[mlir-hlo][cmake] Add dependencies missing for building shared libraries.

This commit adds CMake dependencies that are required for building the
MLIR HLO library with -DBUILD_SHARED_LIBS and currently missing. I am
doing this here
https://github.com/iree-org/iree-llvm-sandbox/blob/b1c16a5/configure.py#L229
through IREE (https://github.com/iree-org/iree).

The reason why I believe the problems did not show up is because
everybody including CI seem to use static linking and not all required
symbols are required for linking static libraries -- they only need to
be present when linking the final executable, where they presumable get
linked in "by chance".

BrianWieder committed:
Patch NCCL to remove an invalid constexpr when compiling with clang using C++17.

It has already been reported to the NCCL Repo: https://github.com/NVIDIA/nccl/issues/759

PiperOrigin-RevId: 503937356

MatthiasKreileder committed:
Clarify that the delegate performance benchmark app bundles a 'mobilenet_quant_v1_224' model statically for latency and accuracy testing.  Remove the misleading README section that suggests that the model is provided on the command line.

PiperOrigin-RevId: 503942517

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@171e7b831227

Updates LLVM usage to match
[171e7b831227](https://github.com/llvm/llvm-project/commit/171e7b831227)

PiperOrigin-RevId: 503958865

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/cd5b944453efe1e7931bd0db08ad8ae370c8adfe.

PiperOrigin-RevId: 503960870

Commit:
Change include of register.h to tensorflow/lite/core/.

Note that only includes from within TF Lite are changed.

PiperOrigin-RevId: 503962161

d0k committed:
[xla:cpu] Fix sign compare warning

https://reviews.llvm.org/D142201 changed the type of rank.

transform_matmul_for_cpu.cc:443:33: error: comparison of integers of different signs: 'int' and 'unsigned long' [-Werror,-Wsign-compare]
              for (int i = 0; i < numLoops; ++i) {
                              ~ ^ ~~~~~~~~

PiperOrigin-RevId: 503973696

BrianWieder committed:
Remove `py_builtins.py`'s dependency on `distribute/input_lib.py` by changing `distribute/input_lib.py` to use the registry instead.

PiperOrigin-RevId: 503978650

sirakiin committed:
Telemetry related changes:

1. InterpreterBuilder also reads all metadata when using C APIs (which uses ::tflite::Model to build interpreter directly instead of FlatBufferModel.
2. For Telemetry C API, TfLiteInterpreterOptionsSetTelemetryProfiler does not take ownership of the profiler struct.

PiperOrigin-RevId: 503984356

Commit:
[KernelGen] Group complex-specific kernels in build file

PiperOrigin-RevId: 503986493

Commit:
Change include of headers to tensorflow/lite/core/.

Note that only includes from within TF Lite are changed.

PiperOrigin-RevId: 503988196

pifon2a committed:
[GmlSt] Use tensor.extract_slice instead of gml_st.materialize.

PiperOrigin-RevId: 503990156

Commit:
[KernelGen] Group arithmetic kernels in build file

PiperOrigin-RevId: 503997133

jblespiau committed:
Add SetPayload(absl::string_view type_url, absl::Cord payload) to map absl.

PiperOrigin-RevId: 503997678

sdasgup3 committed:
[StableHLO to MHLO] Handle Dynamic Bounds in Shape Function of ScatterOp

PiperOrigin-RevId: 503997806

tensorflower-gardener committed:
Merge pull request #58853 from Konstantin-Beluchenko:migrate_fuzzers_to_gft

PiperOrigin-RevId: 503999710

tensorflower-gardener committed:
Slightly relax the test tolerance for Cbrt.

PiperOrigin-RevId: 504005284

tensorflower-gardener committed:
Merge pull request #59418 from snadampal:config_update

PiperOrigin-RevId: 504006190

wilsingosti committed:
#tf-data For autotuning purposes, the number of elements used to compute the self time and the pipeline ratio of a node should be the number of elements produced by the node plus the number of elements in the buffer of the node.

PiperOrigin-RevId: 504009716

ddunl committed:
Delete compiler/xla/stream_executor/lib/mathutil.h and update users to TSL.
Remove DivideCeil from compiler/xla/stream_executor:device_description as it is unused and deprecated.

PiperOrigin-RevId: 504010193

tensorflower-gardener committed:
Simplify update_tpu_metadata's internal implementation.

PiperOrigin-RevId: 504013357

tensorflower-gardener committed:
Merge pull request #59238 from Intel-tensorflow:matmul_fix

PiperOrigin-RevId: 504010808

tensorflower-gardener committed:
Merge pull request #59282 from apach301:fix-hlo_parser

PiperOrigin-RevId: 504014241

hehhjiang committed:
Migrate to TfLiteGpuDelegatePluginCApi symbol from GPU Module.

PiperOrigin-RevId: 504016190

tensorflower-gardener committed:
Merge pull request #59421 from ingomueller-net:mlir-hlo-fix-shared-libs

PiperOrigin-RevId: 504018695

hawkinsp committed:
Fix test failures under NumPy 1.24.

NumPy 1.24 release notes: https://numpy.org/devdocs/release/1.24.0-notes.html

The fixes vary, but there are three particularly common changes:
* NumPy 1.24 no longer automatically infers dtype=object when ragged sequences are passed to np.array(). See https://numpy.org/neps/nep-0034-infer-dtype-is-object.html . In most cases the fix is to pass dtype=object explicitly when forming an array, but in a few cases where the raggedness seems accidental other fixes were used. You may wish to verify that the raggedness is expected.
* NumPy 1.24 removes several previously-deprecated type aliases (np.bool, np.int, np.float, np.complex, np.object, np.str, np.unicode, np.long). This change replaces them with their recommended replacements (bool, int, float, complex, object, str, str, int).
* NumPy 1.24 is pickier about the dtype= option passed to comparison ufuncs.

PiperOrigin-RevId: 504023738

qlzh727 committed:
Add implementation for distribute_datasets_from_function() wrt new mirrored strategy.

PiperOrigin-RevId: 504025606

mikedelorimier committed:
MLIR to graph export allows duplicate control edges. There should be no duplicates in the MLIR that's exported. The duplicate check requires iterating over previously added edges, which is expensive.

PiperOrigin-RevId: 504025957

fcoUnda committed:
Merge branch 'runtime_client_fuzz' of github.com:OleksiiZaderykhinCapgemini/tensorflow into runtime_client_fuzz

gbaned commented:
Hi @fcoUnda Can you please review this PR ? Thank you!

gbaned commented:
Hi @fcoUnda Can you please review this PR ? Thank you!

gbaned commented:
Hi @fcoUnda Can you please review this PR ? Thank you!
